gradle-plugin-testing-junit/src/main/java/com/palantir/gradle/testing/files/FileFactory.java

```diff
@@ -30,6 +30,10 @@ default ArbitraryFile file(String path) {
         return new ArbitraryFile(resolvePath(path));
     }
 
+    default Directory directory(String path) {
+        return new Directory(resolvePath(path)).ensureExists();
```

Comment:
This is just as bad as creating the directories in the constructor! To the user it does the same thing!